app/tests/reader_studies_tests/test_views.py

```diff
@@ -128,3 +128,92 @@ def test_answer_remove(client):
     assert Answer.objects.count() == 1
     assert Answer.objects.filter(creator=r1).count() == 0
     assert Answer.objects.filter(creator=r2).count() == 1
+
+
+@pytest.mark.django_db
+def test_question_delete(client):
+    rs = ReaderStudyFactory()
+    r1, editor = UserFactory(), UserFactory()
+    rs.add_reader(r1)
+    rs.add_editor(editor)
+    q = QuestionFactory(
+        reader_study=rs,
+        question_text="q1",
+        answer_type=Question.ANSWER_TYPE_BOOL,
+    )
+    assert Question.objects.count() == 1
+
+    response = get_view_for_user(
+        viewname="reader-studies:question-delete",
+        client=client,
+        method=client.post,
+        reverse_kwargs={"slug": rs.slug, "pk": q.pk},
+        follow=True,
+        user=r1,
+    )
+
+    assert response.status_code == 403
+
+    response = get_view_for_user(
+        viewname="reader-studies:question-delete",
+        client=client,
+        method=client.post,
+        reverse_kwargs={"slug": rs.slug, "pk": q.pk},
+        user=editor,
+    )
+
+    assert response.status_code == 302
+    assert Question.objects.count() == 0
+    assert str(rs) in response.url
+
+
+@pytest.mark.django_db
+def test_question_delete_disabled_for_questions_with_answers(client):
+    rs = ReaderStudyFactory()
+    r1, editor = UserFactory(), UserFactory()
+    rs.add_reader(r1)
+    rs.add_editor(editor)
+    q = QuestionFactory(
+        reader_study=rs,
+        question_text="q1",
+        answer_type=Question.ANSWER_TYPE_BOOL,
+    )
+    AnswerFactory(creator=r1, question=q, answer=True)
+
+    assert Answer.objects.count() == 1
+    assert Question.objects.count() == 1
+    assert not q.is_fully_editable
+
+    response = get_view_for_user(
+        viewname="reader-studies:question-delete",
+        client=client,
+        method=client.get,
+        reverse_kwargs={"slug": rs.slug, "pk": q.pk},
+        user=editor,
+    )
+
+    assert b"Not possible to delete this question" in response.content
+    assert Question.objects.count() == 1
+
+    # if answer is deleted, deletion of the question is possible again
+    get_view_for_user(
+        viewname="reader-studies:answers-remove",
+        client=client,
+        method=client.post,
+        reverse_kwargs={"slug": rs.slug},
+        data={"user": r1.id},
+        follow=True,
+        user=editor,
+    )
+
+    assert Answer.objects.count() == 0
+
+    response = get_view_for_user(
+        viewname="reader-studies:question-delete",
+        client=client,
+        method=client.get,
+        reverse_kwargs={"slug": rs.slug, "pk": q.pk},
+        user=editor,
+    )
+
```

Comment:
It would also be nice to assert here that the question has actually been removed. I know you already tested that in the other test case, but I still think it would be a good assertion here as well, just in case something weird happens if a question has previously had answers associated with it.

Reply:
The get method that I'm calling doesn't actually delete the question. I would need to use the post method instead for the deletion to succeed. but the post method succeeds in all cases (even if the question has answers), so that didn't make all that much sense. What do you suggest I should do?

Reply:
Ah that's a good point and something I missed. What you could probably do is add a check here: https://github.com/comic/grand-challenge.org/pull/1710/files#diff-d1601f07f94a69bc5526b8feb2501e91db4d928e1bfad9062514905e6a316d65R978

You could just add an if-statement before you add the success message and raise a forbidden if there are any answers for the question. Let me know if you need any help with that.